### PR TITLE
RavenDB-13417 Set exception to the `topologyModification` task

### PR DESCRIPTION
**Pull request by karmeli87**
- If an exception occurred during `TryModifyTopology` we must set the exception to notifiy about it to other waiting threads.
- Don't be eager on update the promotable.

### CODE REVIEW

File: src/Raven.Server/Rachis/Leader.cs [hunk 1]
```diff
@@ -609,7 +609,6 @@ private void CheckPromotables()
 
                 TryModifyTopology(ambassador.Key, ambassador.Value.Url, TopologyModification.Voter, out Task _);
 
-                _promotableUpdated.Set();
```
Comment (ayende) [inline src/Raven.Server/Rachis/Leader.cs:612]: why did you remove this?

Comment (karmeli87) [reply src/Raven.Server/Rachis/Leader.cs:612]: There was a tight loop when the task from `TryModifyTopology` wasn't completed.

**Review by ayende — APPROVED** [review]
